Packs/SophosCentral/Integrations/SophosCentral/SophosCentral.py

```diff
@@ -2022,6 +2027,17 @@ def sophos_central_endpoint_list_command(client: Client, args: dict) -> CommandR
     if items:
         for item in items:
             object_data = {field: item.get(field) for field in table_headers}
+            # The Sophos API does partial string matching, which can be unwanted for IP addresses.
+            if full_match_ip:
+                query_ips = argToList(args.get("ip_addresses"))
+                ips = []
+                if (data := object_data.get("ipv4Addresses", [])) is not None:
+                    ips.extend(data)
```

Comment:
The check `is not None` is redundant because `object_data.get("ipv4Addresses", [])` will always return a list, either the value associated with the key "ipv4Addresses" exists, or an empty list otherwise.
You can make the code more concise and readable like so:
```suggestion
                ips.extend(object_data.get("ipv4Addresses", []))
```

Reply:
I would say the check isn't redundant depending on the `object_data` that is returned by the API. Consider the `object_data` to look like this:

```
object_data = {
    "ip_addresses": None
}
```

Extending the `ips` will then lead to a `TypeError`.

Reply:
Gotcha, so in that case your version is perfectly logical. No need for changes here.